internal/productcatalog/feature_connector.go

```diff
@@ -0,0 +1,151 @@
+package productcatalog
+
+import (
+	"context"
+	"slices"
+
+	"github.com/openmeterio/openmeter/internal/meter"
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+type CreateFeatureInputs struct {
+	Name                string             `json:"name"`
+	Key                 string             `json:"key"`
+	Namespace           string             `json:"namespace"`
+	MeterSlug           string             `json:"meterSlug"`
+	MeterGroupByFilters *map[string]string `json:"meterGroupByFilters"`
+}
+
+type FeatureConnector interface {
+	// Feature Management
+	CreateFeature(ctx context.Context, feature CreateFeatureInputs) (Feature, error)
+	// Should just use deletedAt, there's no real "archiving"
+	ArchiveFeature(ctx context.Context, featureID models.NamespacedID) error
+	ListFeatures(ctx context.Context, params ListFeaturesParams) ([]Feature, error)
+	GetFeature(ctx context.Context, featureID models.NamespacedID) (Feature, error)
+}
+
+type FeatureOrderBy string
+
+const (
+	FeatureOrderByCreatedAt FeatureOrderBy = "created_at"
+	FeatureOrderByUpdatedAt FeatureOrderBy = "updated_at"
+)
+
+type ListFeaturesParams struct {
+	Namespace       string
+	IncludeArchived bool
+	Offset          int
+	Limit           int
+	OrderBy         FeatureOrderBy
+}
+
+type DBCreateFeatureInputs struct {
+	Name                string
+	Key                 string
+	Namespace           string
+	MeterSlug           string
```

Comment:
Shouldn't this be a pointer as meter is optional?

Reply:
well yes, but i'd push that to next PR, currently it isn't (as it wasn't previously)